File: receipt-analyzer-service/src/main/java/pl/madzierski/daniel/app/receipt/revision/RevisionProvider.java
package pl.madzierski.daniel.app.receipt.revision;

import lombok.AllArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
@AllArgsConstructor
public class RevisionProvider {

    private final ReceiptRevisionRepository revisionRepository;

    public List<ReceiptRevisionEntity> getReceiptRevisions(String receiptId) {
        return revisionRepository.findReceiptRevisionEntityByReceiptId(receiptId);
    }
}

File: receipt-analyzer-service/src/main/kotlin/pl/madzierski/daniel/app/receipt/revision/RevisionProvider.kt
package pl.madzierski.daniel.app.receipt.revision

import org.springframework.stereotype.Service

@Service
class RevisionProvider(val revisionRepository: ReceiptRevisionRepository) {
    fun getReceiptRevisions(receiptId: String): List<ReceiptRevisionEntity> =
        revisionRepository.findReceiptRevisionEntityByReceiptId(receiptId)

}